### PR TITLE
Add Pkl language

### PR DESCRIPTION
**Pull request by stackoverflow**
## Description

Adds support for the [Pkl](https://pkl-lang.org) programming language.

## Checklist:

- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      -  https://github.com/search?type=code&q=NOT+is%3Afork+path%3A*.pkl+%2F%5E%5Cs*%28module%7Cimport%5Cs*%22%7Camends%7Cextends%7Clocal%7Cconst%7Cfixed%7Cabstract%7Copen%5Cs*%5Bclass%7Cmodule%5D%7Cclass%5Cs*%5B%5E%3D%5D%7Ctypealias%7C%40%5Cw%2B%29%5Cb%2F+NOT+user%3Aapple
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - https://github.com/apple/pkl-pantry/blob/main/packages/org.json_schema/Parser.pkl
      - https://github.com/apple/pkl-pantry/blob/main/packages/io.prometheus/examples/rule.pkl
      - https://github.com/apple/pkl-pantry/blob/main/packages/org.openapis.v3/tests/SchemaGenerator.pkl
    - Sample license(s):
      - Apache-2.0
  - [x] I have included a syntax highlighting grammar: https://github.com/apple/pkl.tmbundle
      
  - [x] I have added a color
    - Hex value: `#6b9543`
    - Rationale: This is one of the colors of the [Pkl logo](https://pkl-lang.org/_/img/favicon.svg)
  - [x] I have updated the heuristics to distinguish my language from others using the same extension.

### CODE REVIEW

**Comment by stackoverflow** [comment]
I've tried to remove our org from the language search but I'm not sure the syntax supports it.

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -5240,6 +5240,19 @@ Pike:
   tm_scope: source.pike
   ace_mode: text
   language_id: 287
+Pkl:
+  type: programming
+  color: "#6b9543"
+  extensions:
+  - ".pkl"
+  - ".pcf"
```
Comment (lildude) [inline lib/linguist/languages.yml:5248]: We need samples and supporting searches for all extensions and filenames added to a PR. If usage isn't sufficient, support can be added later and an override can be used in the interim.

Also keep in mind that the same grammar and thus syntax highlighting will be applied to all extensions and filenames associated with a language.

**Review by lildude — CHANGES_REQUESTED** [review]
See inline comment.

> I've tried to remove our org from the language search but I'm not sure the syntax supports it.

Sure does. Add `NOT user:apple` to the query.

File: lib/linguist/languages.yml (hunk 1 above)
Comment (stackoverflow) [reply lib/linguist/languages.yml:5248]: > We need samples and supporting searches for all extensions and filenames added to a PR. If usage isn't sufficient, support can be added later and an override can be used in the interim.

Oh, I guess I missed that and thought they could be lumped together as they are just different extensions/names for the same language.
What do you mean by `an override can be used in the interim.`?

> Also keep in mind that the same grammar and thus syntax highlighting will be applied to all extensions and filenames associated with a language.

That is fine. Both `.pcf` and `PklProject` are just Pkl files. `pcf` is a subset of Pkl.

Comment (lildude) [reply lib/linguist/languages.yml:5248]: > What do you mean by `an override can be used in the interim.`?

https://github.com/github-linguist/linguist/blob/master/docs/overrides.md

**Comment by stackoverflow** [comment]
@lildude I removed `.pcf` and `PklProject` from the list of file names and extensions. We don't have the expected usage data yet. Will add this later as a separate PR once we do have the data.

For some reason my last commit didn't appear in the PR even though it's there if you click the branch...

**Comment by stackoverflow** [comment]
Hi @lildude. Are we still pending on popularity? There's considerably more pkl files in the search since I created this PR.

**Comment by Senjai** [comment]
Just commenting for context RE: Popularity, we're currently using this in one or two of Shopify's internal repositories. I'd like to pitch using this for some of our external open source stuff, but there is a bit of chicken-and-egg there since we want the Github experience to be fantastic for our open source projects as well. Just adding another viewpoint if it helps at all.

@lildude Popularity aside, are you generally okay with the implementation here?

**Comment by hugolgst** [comment]
any updates? would love to see it move forward.

**Review by lildude — APPROVED** [review]

**Comment by hugolgst** [comment]
amazing!!

**Pull request merged**